crates/fuel-core/src/service/adapters/graphql_api.rs

```diff
@@ -180,3 +193,42 @@ impl ConsensusProvider for ConsensusParametersProvider {
         self.shared_state.latest_consensus_parameters()
     }
 }
+
+impl ImportResultProvider for ImportResultProviderAdapter {
+    fn result_at_height(
+        &self,
+        height: Option<BlockHeight>,
```

Comment:
This is confusing. Maybe create a new type:
```
enum BlockHeightQuery {
    Specific(BlockHeight),
    Genesis,
}
```
Same cardinality, but easier to understand.

Reply:
This kind of change is not for this PR=) We use `None` as an identifier of the genesis in a lot of places.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2005